web/src/lib/components/user-settings-page/PincodeSettings.svelte

```diff
@@ -0,0 +1,137 @@
+<script lang="ts">
+  import {
+    notificationController,
+    NotificationType,
+  } from '$lib/components/shared-components/notification/notification';
+  import PincodeInput from '$lib/components/user-settings-page/PincodeInput.svelte';
+  import { changePincode, createPincode, getAuthStatus } from '@immich/sdk';
+  import { Button } from '@immich/ui';
+  import type { HttpError } from '@sveltejs/kit';
+  import { onMount } from 'svelte';
+  import { t } from 'svelte-i18n';
+  import { fade } from 'svelte/transition';
+
+  let pincodeFormElement = $state<HTMLFormElement | null>(null);
+  let hasPincode = $state(false);
+  let currentPincode = $state('');
+  let newPincode = $state('');
+  let confirmPincode = $state('');
+  let isLoading = $state(false);
+
+  const onSubmit = async (event: Event) => {
+    event.preventDefault();
+
+    if (hasPincode) {
+      await handleChangePincode();
+      return;
+    }
+
+    await handleCreatePincode();
+  };
+
+  onMount(async () => {
+    const authStatus = await getAuthStatus();
+    hasPincode = authStatus.hasPincode;
+  });
+
+  const canSubmit = $derived(
+    (hasPincode ? currentPincode.length === 6 : true) &&
+      newPincode.length === 6 &&
+      confirmPincode.length === 6 &&
+      newPincode === confirmPincode,
```

Comment:
If they're the same they also need to have the same length anyways.

Reply:
Yes, but we want the input to have length 6.

Reply:
Then it's sufficient if you just check either of them, right?